### PR TITLE
fixes to time series mapping file generation

### PR DESCRIPTION
**Pull request by Juliamg**
## Description
- Fix script that creates new time_series_mapping config
- Alter transformations translation script so that it can be run from customer repos
- Generate a file with all new and unique transformations (for testing/sanity check purposes)

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/power-ops-sdk/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [pyproject.toml](https://github.com/cognitedata/power-ops-sdk/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: tests/data/generated/transformations_v2.yaml [hunk 1]
```diff
@@ -13,582 +13,17 @@ transformations:
     parameters:
       relative_datapoints:
       - offset_minute: 0.0
-        offset_value: 100000.0
-- StaticValues:
```
Comment (doctrino) [inline tests/data/generated/transformations_v2.yaml:17]: Earlier bug?

File: scripts/transformationsv2_config_translations.py [hunk 2]
```diff
@@ -1,36 +1,71 @@
...
 from pathlib import Path
 
 import yaml
 
 from cognite.powerops.resync.config import ReSyncConfig, Transformation
 from cognite.powerops.resync.models._shared_v1_v2.cogshop_model import transformations_v2_transformer
 
-REPO_ROOT = Path(__file__).resolve().parent.parent
+
+def immutify_dictionary(d):
```
Comment (doctrino) [inline scripts/transformationsv2_config_translations.py:15]: type hints?

Comment (doctrino) [reply scripts/transformationsv2_config_translations.py:15]: This seems to also unpack `numpy` and `pandas` structures. Suggest returning a `MappingProxy`. See https://stackoverflow.com/questions/32720492/why-is-a-class-dict-a-mappingproxy

File: scripts/transformationsv2_config_translations.py [hunk 3]
```diff
@@ -1,36 +1,71 @@
...
+            if hasattr(v, "dtype"):
+                d_new[k] = v.item()
+            else:
+                d_new[k] = v
+
+    return dict(sorted(d_new.items(), key=lambda item: item[0]))
+
+
+def hash_dictionary(d):
+    d_hashable = immutify_dictionary(d)
```
Comment (doctrino) [inline scripts/transformationsv2_config_translations.py:34]: Is this step necessary? I would expect you could only hash the json string and be good.

**Review by doctrino — APPROVED** [review]
Just a few comments.

File: scripts/transformationsv2_config_translations.py (hunk 2 above)
Comment (doctrino) [reply scripts/transformationsv2_config_translations.py:15]: Maybe rename to `to_sorted_hashable_dictionary`

File: tests/data/generated/transformations_v2.yaml (hunk 1 above)
Comment (Juliamg) [reply tests/data/generated/transformations_v2.yaml:17]: Rather that only the unique ones are written to this file now, so duplicates are removed. Also, this file is not used by resync in the end (only the new time series mappings and price scenarios files). It is just something I added to be able to be able to do some verification and sanity checks

**Pull request merged**